src/main/java/javaslang/render/text/Box.java

```diff
@@ -3,130 +3,72 @@
 import javaslang.collection.List;
 import javaslang.collection.Stream;
 
-import java.io.Serializable;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Objects;
 
-public final class Box implements Serializable {
+public abstract class Box {
 
-    private static final long serialVersionUID = 0L;
+    public enum HAlignment {LEFT, CENTER, RIGHT}
 
-    private final String[] contents;
+    public enum VAlignment {TOP, CENTER, BOTTOM}
 
-    private Box(String[] contents) {
-        this.contents = contents;
-    }
-
-    public static Box of(String string) {
-        Objects.requireNonNull(string, "string is null");
-        List<String> list = List.of(string.split("\n"));
-        int w = list.map(String::length).max().get();
-        String[] contents = list.map(s -> {
-            int n = s.length();
-            return replicate(' ', (w - n) / 2) + s + replicate(' ', w - n - (w - n) / 2);
-        }).toJavaArray(String.class);
-        return new Box(contents);
-    }
-
-    public static Box of(String... s) {
-        return new Box(s);
-    }
-
-    public static Box empty() {
-        return new Box(new String[0]);
-    }
+    protected int w;
+    protected int h;
```

Comment:
I would make these final and provide a (protected) constructor that is called by subclassed.